libkbfs/kbfs_ops_concur_test.go

```diff
@@ -1226,6 +1226,105 @@ func TestKBFSOpsMultiBlockWriteDuringRetriedSync(t *testing.T) {
 	}
 }
 
+// This tests the situation where cancellation happens when MD writes has
+// started, and cancellation is delayed. Since no extra delay greater than the
+// grace period in MD writes is introduced, Create should succeed.
+func TestKBFSOpsCanceledCreateNoError(t *testing.T) {
+	config, _, _ := kbfsOpsConcurInit(t, "test_user")
+	defer CheckConfigAndShutdown(t, config)
+
+	ctx := context.Background()
+
+	onPutStalledCh, putUnstallCh, ctx :=
+		StallMDOp(ctx, config, StallableMDPut)
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	ctx, err := NewContextWithCancellationDelayer(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootNode := GetRootNodeOrBust(t, config, "test_user", false)
+
+	kbfsOps := config.KBFSOps()
+	errChan := make(chan error)
+	go func() {
+		_, _, err := kbfsOps.CreateFile(ctx, rootNode, "a", false, WithExcl)
+		errChan <- err
+	}()
+
+	// Wait until Create gets stuck at MDOps.Put(). At this point, the delayed
+	// cancellation should have been enabled.
+	<-onPutStalledCh
+	cancel()
+	close(putUnstallCh)
+
+	// We expect no canceled error
+	err = <-errChan
+	if err != nil {
+		t.Fatalf("Create returned error: %v", err)
+	}
+
+	// SyncFromServerForTesting and make sure it worked.
+	err = kbfsOps.SyncFromServerForTesting(ctx, rootNode.GetFolderBranch())
```

Comment:
It's a bit weird to be using the cancelled `ctx` here.  Also, this shouldn't be necessary because the previous operation succeeded, so I would just get rid of this call.